omod/src/main/webapp/resources/scripts/visit/visit.js

```diff
@@ -631,6 +631,23 @@ angular.module("visit", [ "filters", "constants", "encounterTypeConfig", "visitS
                 });
             }
 
+            function loadAllPatientEncounters(patientUuid) {
+                Encounter.get({
+                    patient: patientUuid,
+                    encounterType: encounterTypeUuid ? encounterTypeUuid : '',
+                    order: 'desc',
+                    v: "custom:(encounterDatetime,encounterType:(uuid,display))"
```

Comment:
I've made it so that if 500 encounters are fetched, where 500 is presumed to be the default limit, it will query the maximum limit and fetch that many. It seems that the default maximum is 3000.

This method has a slight disadvantage for patients with >500 encounters, adding ~10% to page load time in my testing. But it doesn't penalize patients with <500 encounters.

Reply:
It's a little funny so I'd like a review of this before I merge it in.

Reply:
I would have expected that the REST API would provide some paging information back if it is limiting results, is that not true?

Reply:
I was about to reply that there was no such thing, and then I looked into it and realized that the `links` object, which I'm not in the habit of noticing, has exactly that. Refactored.

Reply:
Right, sorry, I should have added that there is some paging functionality available, makes the most sense to use that.